src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/CreateReportCommandHandler.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Esfa.Recruit.Vacancies.Client.Application.Commands;
+using Esfa.Recruit.Vacancies.Client.Domain.Events;
+using Esfa.Recruit.Vacancies.Client.Domain.Messaging;
+using Esfa.Recruit.Vacancies.Client.Domain.Repositories;
+using MediatR;
+using System.Threading;
+using System.Threading.Tasks;
+using Esfa.Recruit.Vacancies.Client.Application.Providers;
+using Esfa.Recruit.Vacancies.Client.Domain.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace Esfa.Recruit.Vacancies.Client.Application.CommandHandlers
+{
+    public class CreateReportCommandHandler : IRequestHandler<CreateReportCommand>
+    {
+        private readonly ILogger<CreateReportCommandHandler> _logger;
+        private readonly IReportRepository _repository;
+        private readonly IMessaging _messaging;
+        private readonly ITimeProvider _timeProvider;
+
+        public CreateReportCommandHandler(
+            ILogger<CreateReportCommandHandler> logger,
+            IReportRepository repository, 
+            IMessaging messaging, 
```

Comment:
In there for future use?

Reply:
That was the plan but in the end I'm not using messaging so I have now removed.